src/main/java/frc/robot/commands/DistanceSensorDetectorCommand.java

```diff
@@ -0,0 +1,51 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018-2019 FIRST. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
+import com.revrobotics.*;
+import com.revrobotics.Rev2mDistanceSensor.Port;
+import com.revrobotics.Rev2mDistanceSensor.Unit;
+
+
+/**
+ * Add your docs here.
+ */
+public class DistanceSensorDetectorCommand extends CommandBase{
```

Comment:
This whole thing should be a subsystem, not a Command.  I think this is my fault for telling you it should be a Command.  I understand better now how to architect using Commands and Subsystems.

Basically, it needs to stop extending CommandBase, and instead extend SubsystemBase.

Change the execute() method to be periodic().

You'll need to change the name of the class to get rid of Command (and change the name of the constructor, and move it to the Subsytems directory.

Reply:
Also, you'll want to add the subsystem to the ShooterCommand constructor, pass it in, and make sure that you add this DistanceSensorDetector as a requirement in that command.